### PR TITLE
[VSTS] Make the pipeline build and notarize the pkgs

### PR DESCRIPTION
**Pull request by mandel-macaque**
First steps for the migration, the job only does the following:

1. Builds the project
2. Builds the pkgs
3. Builds the nugets
4. Signs the pkgs
5. Uploads everything as a build artifact.

Next steps should be: 

1. Upload to azure blob storage. 
2. Create the artifacts json
3. Add Github status notifications.

### CODE REVIEW

**Review by mjbond-msft — APPROVED** [review]
As the initial iteration to build, package & sign, this looks good

File: tools/devops/provision-brew-packages.csx [hunk 1]
```diff
@@ -1 +1 @@
-BrewPackages ("cmake", "autoconf", "automake", "libtool", "python3", "p7zip");
\ No newline at end of file
+BrewPackages ("cmake", "autoconf", "automake", "libtool", "p7zip", "libmagic");
```
Comment (tj-devel709) [inline tools/devops/provision-brew-packages.csx:1]: Perhaps new line at end in next iteration of pipeline changes!

**Review by tj-devel709 — APPROVED** [review]

**Review by spouliot — COMMENTED** [review]
❤️

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-10058/1/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-10058/1/display/redirect) (on internal Jenkins) failed in stage(s) 'Checkout' 🔥 : hudson.AbortException: script returned exit code 128

**Pull request merged**